src/file_reader.h

```diff
@@ -4,7 +4,7 @@
 
 char *read_file(const char *file_name) {
     FILE *fp;
-    fp = fopen(file_name, "r");
+    fp = fopen(file_name, "rb+");
```

Comment:
On my Windows machine "r" did not work, since it counted "\r\n" as two seperate chars, which made the sequence length not match the expected. After we changed it to "rb+" it worked on both Linux and Windows. The "+" is just a result of using a solution that worked, without knowing the full difference between "rb" and "rb+".